src/main/scala/strawman/collection/immutable/SortedSet.scala

```diff
@@ -0,0 +1,21 @@
+package strawman.collection.immutable
+
+import strawman.collection.{SetMonoTransforms, Sorted, SortedLike}
+
+/** Base trait for sorted sets */
+trait SortedSet[A]
+  extends Set[A]
+    with Sorted[A]
+    with SortedSetLike[A, SortedSet] // Inherited SortedSet operations return a `SortedSet`
+
+trait SortedSetLike[A, +C[X] <: SortedSet[X]]
+  extends SortedLike[A, C]
+    with SetLike[A, Set] // Inherited Set operations return a `Set`
+    with SetOps[A, C[A]] // Override the return type of Set ops to return C[A]
+    with SetMonoTransforms[A, C[A]] {
```

Comment:
There is subtlety here: we inherit from `SetLike[A, Set]` so that methods like `map` return a `Set` rather than a `SortedSet`. However we don’t want `+` or `++` to return a `Set` but a `SortedSet`, hence the `SetOpt[A, C[A]]` and `SetMonoTransforms[A, C[A]]` mixins.